### PR TITLE
Extract EvmTestRunner to run common_tests

### PR DESCRIPTION
**Pull request by MattMSumner**
Why:

We would like to start profiling our evm code to look for areas where we
can improve performance. [Parity] has a [benchmark script] for
outputting how long each of the evm common_tests take to run

[Parity]: https://github.com/paritytech/parity-ethereum
[benchmark script]: https://github.com/paritytech/parity-ethereum/blob/f6dcca3ebb6341049b8fc4c890a463277c493bc4/scripts/evm_jsontests_bench.sh

This PR:

Extracts EvmTestRunner. This allows us to run the following type of code
for profiling:

```
% cd apps/evm
% MIX_ENV=test iex -S mix

iex(1)> test_json =
...(1)> File.read!("../../ethereum_common_tests/VMTests/vmPerformance/ackermann31.json") |>
...(1)> Jason.decode!() |>
...(1)> Map.get("ackermann31")
iex(2)> :fprof.apply(EvmTestRunner, :run, [test_json])
iex(3)> :fprof.profile
iex(4)> :fprof.analyse([callers: true, sort: :own, totals: true, details: true])
```

### CODE REVIEW

**Comment by InoMurko** [comment]
whooaaa, this is great! do you think it would make sense to run this separately or after the tests in circle?
this way we could monitor potential performance degradations. wdyt?

File: apps/evm/test/support/evm_test_runner.ex [hunk 1]
```diff
@@ -0,0 +1,110 @@
+defmodule EvmTestRunner do
+  import ExthCrypto.Math, only: [hex_to_bin: 1, hex_to_int: 1]
+
+  alias EVM.Mock.{MockAccountRepo, MockBlockHeaderInfo}
+
+  def run(json) do
```
Comment (germsvel) [inline apps/evm/test/support/evm_test_runner.ex]: Not sure if this is desired, but I _think_ the other test runners accept a test name and retrieve the file. We could do that to keep in line with them, and it makes it easier to reuse the test runner with a rake task when we only wan to run a single test. What do you think?

File: apps/evm/test/support/evm_test_runner.ex [hunk 2]
```diff
@@ -0,0 +1,110 @@
+defmodule EvmTestRunner do
```
Comment (InoMurko) [inline apps/evm/test/support/evm_test_runner.ex]: is this the correct namespace?

File: apps/evm/test/support/evm_test_runner.ex [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  import ExthCrypto.Math, only: [hex_to_bin: 1, hex_to_int: 1]
+
+  alias EVM.Mock.{MockAccountRepo, MockBlockHeaderInfo}
+
+  def run(json) do
+    exec_env = get_exec_env(json)
+    gas = hex_to_int(json["exec"]["gas"])
+    EVM.VM.run(gas, exec_env)
```
Comment (InoMurko) [inline apps/evm/test/support/evm_test_runner.ex]: could we alias this?

File: apps/evm/test/support/evm_test_runner.ex [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  end
+
+  defp get_exec_env(json) do
+    %EVM.ExecEnv{
```
Comment (InoMurko) [inline apps/evm/test/support/evm_test_runner.ex]: and this too, credo will nag about it

**Comment by andogro** [comment]
Any clarifications or additional instructions needed in test section of README (or additions to EVM README) for this change?

File: apps/evm/test/support/evm_test_runner.ex (hunk 1 above)
Comment (MattMSumner) [reply apps/evm/test/support/evm_test_runner.ex]: I like the idea of doing this for the consistency across test runners. I would like to keep reading files out of the is test_runner to eliminate it as a factor when benchmarking/running profiling. How would you feel if I opened a PR to remove reading files from the other test_runners?

**Comment by MattMSumner** [comment]
@andogro not at this time. This is a small refactor to allow me to manually do some profiling. Once I've learnt more around what is useful to look at (and got all the performance tests running in a reasonable time) I'll likely a test to check for performance degradation as @InoMurko suggested. That will likely require some documentation changes.

File: apps/evm/test/support/evm_test_runner.ex (hunk 2 above)
Comment (MattMSumner) [reply apps/evm/test/support/evm_test_runner.ex]: Thanks, changed to `EVM.TestRunner`

**Pull request merged**